cpp/src/parquet/arrow/reader.h

```diff
@@ -303,6 +303,9 @@ class PARQUET_EXPORT FileReader {
   /// Set number of records to read per batch for the RecordBatchReader.
   virtual void set_batch_size(int64_t batch_size) = 0;
 
+  /// Set whether to enable smallest decimal arrow type
+  virtual void set_smallest_decimal_enabled(bool smallest_decimal_enabled) = 0;
```

Comment:
This looks weird. We cannot change it after the reader has been created. Isn't it accessible via the ArrowReaderProperties?